pkg/applications/installer_test.go

```diff
@@ -0,0 +1,293 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package applications
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	semverlib "github.com/Masterminds/semver/v3"
+
+	appkubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/apps.kubermatic/v1"
+	kubermaticlog "k8c.io/kubermatic/v2/pkg/log"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	fakectrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestApplicationManager_applyNamespace_with_create_ns(t *testing.T) {
```

Comment:
I have to say I am not a fan of the multi-mixed Camel and Snake case here. I see that we do it sometimes in other tests as well. However in these cases (e.g. https://github.com/kubermatic/kubermatic/blob/master/pkg/api/v1/types_test.go#L29) it's usually only a single underscore. This would be an okay compromise for me :) 
So TestApplicationManager_ApplyNamespaceWithCreateNs and so on